pkg/infrastructure/aws/clusterapi/aws.go

```diff
@@ -146,9 +146,6 @@ func (*Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput)
 		return fmt.Errorf("failed to find HostedZone ID for NLB: %w", err)
 	}
 	apiHost := awsCluster.Status.Network.SecondaryAPIServerELB.DNSName
-	if awsCluster.Status.Network.APIServerELB.Scheme == capa.ELBSchemeInternetFacing {
-		apiHost = awsCluster.Status.Network.APIServerELB.DNSName
-	}
 	apiIntHost := awsCluster.Spec.ControlPlaneEndpoint.Host
 	err = client.CreateOrUpdateRecord(ctx, in.InstallConfig.Config, apiHost, apiIntHost, phzID, aliasZoneID)
```

Comment:
this looks like its going to attempt to make a record for a public zone even when the install is internal. Is this true or did I miss something?